File: __aacontest/2.py
class Solution:
    def countSubstrings(self, s: str, c: str) -> int:
        count = 0
        char_count = s.count(c)
        
        # Calculate the number of substrings starting and ending with 'c'
        return char_count * (char_count + 1) // 2

test = Solution()
print(test.countSubstrings("abada", "a"))  # Output: 6
print(test.countSubstrings("zzz", "z"))    # Output: 6

File: __aacontest/2.rb
def count_substrings(s, c)
  count = 0
  s.each_char.with_index do |char, index|
    count += 1 if char == c
  end
  count
end

# Example test cases
puts count_substrings("abada", "a") # Output: 6
puts count_substrings("zzz", "z")   # Output: 6